src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -399,17 +399,33 @@ public void validate(FailureCollector collector) {
 
     //This method validated the pattern of CMEK Key resource ID.
     private void validateCmekKey(FailureCollector failureCollector) {
-      String[] part = cmekKey.split("/");
-      if ((part.length != 8) || !part[0].equals("projects") || !part[2].equals("locations")
-        || !part[4].equals("keyRings") || !part[6].equals("cryptoKeys")) {
-        failureCollector.addFailure("Invalid cmek key resource Id format. It should be of the form " +
-                                      "projects/<gcp-project-id>/locations/<key-location>/keyRings/" +
-                                      "<key-ring-name>/cryptoKeys/<key-name>", null)
-          .withConfigProperty(NAME_CMEK_KEY);
-      } else if (!part[3].equals(location)) {
-        failureCollector.addFailure(String.format("Cloud KMS region '%s' not available for use " +
-                                                    "with GCS region '%s'.", part[3], location), null)
-          .withConfigProperty(NAME_CMEK_KEY);
+      try {
+        CryptoKeyName cmekKeyName = CryptoKeyName.parse(cmekKey);
+        Boolean isServiceAccountFilePath = isServiceAccountFilePath();
+        if (isServiceAccountFilePath != null) {
+          Credentials credentials = getServiceAccount() == null ?
```

Comment:
is the first getServiceAccount() == null needed? It gets passed to loadServiceAccountCredentials() already.